src/miner.cpp

```diff
@@ -534,7 +555,11 @@ void static BitcoinMiner(CWallet *pwallet)
                     SetThreadPriority(THREAD_PRIORITY_NORMAL);
                     LogPrintf("ZcashMiner:\n");
                     LogPrintf("proof-of-work found  \n  hash: %s  \ntarget: %s\n", pblock->GetHash().GetHex(), hashTarget.GetHex());
-                    ProcessBlockFound(pblock, *pwallet, reservekey);
+                    if (ProcessBlockFound(pblock, *pwallet, reservekey)) {
+                        // Ignore chain updates caused by us
+                        std::lock_guard<std::mutex> lock{m_cs};
+                        cancelSolver = false;
```

Comment:
There's a race condition where a chain update (not by us) happens between the one caused by ``ProcessBlockFound`` and taking the lock. It doesn't affect correctness; it will just result in a stale block, if we're really unlucky. I don't think that's a problem. It might be fixable by taking the lock before calling ``ProcessBlockFound``, but that would only work if locks are reentrant.